src/coreclr/jit/llvm.cpp

```diff
@@ -2643,7 +2690,30 @@ void Llvm::lowerToShadowStack()
         _currentRange = &LIR::AsRange(_currentBlock);
         for (GenTree* node : CurrentRange())
         {
-            if (node->OperIs(GT_STORE_LCL_VAR, GT_LCL_VAR, GT_LCL_FLD, GT_LCL_VAR_ADDR, GT_LCL_FLD_ADDR))
+            if (node->OperIs(GT_STORE_LCL_VAR, GT_LCL_VAR, GT_STORE_LCL_FLD, GT_LCL_FLD_ADDR, GT_LCL_VAR_ADDR))
+            {
+                // fail unsupported promoted structs
```

Comment:
These will be dependently promoted, so we don't need to do anything special about them. We do need to rewrite nodes referencing the fields to reference the parent, however.